lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -1229,7 +1275,37 @@ func (d *zfs) RefreshVolume(vol Volume, srcVol Volume, srcSnapshots []Volume, al
 
 // DeleteVolume deletes a volume of the storage device. If any snapshots of the volume remain then
 // this function will return an error.
+// For image volumes, both filesystem and block volumes will be removed.
 func (d *zfs) DeleteVolume(vol Volume, op *operations.Operation) error {
+	for _, suffix := range blockBackedAllowedFilesystems {
```

Comment:
The use of `suffix` as the loop iteration name is somewhat confusing. We don't know at this point as we are reading it what the `suffix` will be used for, so its somewhat abstract. A clearer variable name in my opinion would be `filesystem` as that is what the variable represents. By the time I got further down this function to where `suffix` is used I'd forgotten that it contained the filesystem name. I can see by the Sprintf pattern that `suffix` is being used as a suffix for the volume name, but I would have found it more readable if the variable was called filesystem as then I wouldn't have had to scroll back to the top of the function. Does that make sense?